retrofit/src/test/java/retrofit/RequestBuilderTest.java

```diff
@@ -1008,22 +1009,22 @@ Response method(
     Request request = buildRequest(Example.class, params);
     assertThat(request.method()).isEqualTo("GET");
     assertThat(request.headers().size()).isZero();
-    assertThat(request.urlString()).isEqualTo("http://example.com/foo/bar/?k+it=k%20t&pi+ng=p%20g");
+    assertThat(request.urlString()).isEqualTo("http://example.com/foo/bar/?k%20it=k%20t&pi%20ng=p%20g");
     assertThat(request.body()).isNull();
   }
 
   @Test public void normalPostWithPathParam() {
     class Example {
       @POST("/foo/bar/{ping}/") //
-      Response method(@Path("ping") String ping) {
+      Response method(@Path("ping") String ping, @Body Object body) {
```

Comment:
Retrofit can coerce the absence of body into the empty body.

On Mon, Jun 1, 2015 at 12:27 PM Roman Mazur <notifications@github.com>
wrote:

> In retrofit/src/test/java/retrofit/RequestBuilderTest.java
> <https://github.com/square/retrofit/pull/848#discussion_r31442796>:
>
> >      assertThat(request.body()).isNull();
> >    }
> >
> >    @Test public void normalPostWithPathParam() {
> >      class Example {
> >        @POST("/foo/bar/{ping}/") //
> > -      Response method(@Path("ping") String ping) {
> > +      Response method(@Path("ping") String ping, @Body Object body) {
>
> So is it a deliberate decision and you want to prevent declaration of
> @POST methods without @Body? Or would you accept a change that revives
> such declarations?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/square/retrofit/pull/848/files#r31442796>.
>